export.py

```diff
@@ -2350,6 +2352,9 @@ def export_render_settings(ri, rpass, scene, preview=False):
     ri.PixelFilter(rm.pixelfilter, rm.pixelfilter_x, rm.pixelfilter_y)
     ri.ShadingRate(rm.shadingrate)
     ri.Attribute("trace", depths)
+    if rm.use_separate_path_depths:
+        ri.Option("trace", {'string depthmode': 'separate'})
```

Comment:
I wasn't sure if the option would work earlier in the rib but sure it can be moved there if needed. It does work in this location. Also being that it's a scene variable it will only be output once.  I noticed all the other trace parameters were 'attributes' instead of 'options' so that's why it is by itself.

Sent from my iPhone

On Jun 30, 2016, at 5:00 PM, bsavery <notifications@github.com<mailto:notifications@github.com>> wrote:


In export.py<https://github.com/bsavery/PRMan-for-Blender/pull/295#discussion_r69218182>:

> @@ -2350,6 +2352,9 @@ def export_render_settings(ri, rpass, scene, preview=False):
>      ri.PixelFilter(rm.pixelfilter, rm.pixelfilter_x, rm.pixelfilter_y)
>      ri.ShadingRate(rm.shadingrate)
>      ri.Attribute("trace", depths)
> +    if rm.use_separate_path_depths:
> +        ri.Option("trace", {'string depthmode': 'separate'})


So first of all you're going to output an option multiple times postentially. Which doesn't really matter here but generally a weird idea. And options should really be output earlier in the rib with the others.... Maybe when options are output could scan through and see if any have relative pixel variance set?

-
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub<https://github.com/bsavery/PRMan-for-Blender/pull/295/files/ebc2c9bc75101bf98114a1e08b377171093b6e6e#r69218182>, or mute the thread<https://github.com/notifications/unsubscribe/AM5RFO2J9dh6tmQFY9ED2y_WUil6ZRakks5qRDxxgaJpZM4JCeTu>.